Source/JavaScriptCore/runtime/JSType.h

```diff
@@ -127,6 +127,7 @@ enum JSType : uint8_t {
     JSWeakSetType,
     WebAssemblyModuleType,
     WebAssemblyInstanceType,
+    WebAssemblyGCObjectType,
```

Comment:
I wish we could avoid adding a new type here

Reply:
Yeah unfortunately I'm not aware of a way to avoid it. The C++ code can just use `jsDynamicCast` (https://github.com/WebKit/WebKit/blob/60ff12d314ad2127a1ff211c05636ff630f29582/Source/JavaScriptCore/wasm/WasmOperationsInlines.h#L300) but I don't think it's easy to use this generic dispatch from JIT.